### PR TITLE
noUncheckedIndexedAccess: true

### PR DESCRIPTION
**Pull request by dcsaszar**
https://github.com/Scrivito/scrivito-portal-app/pull/329#discussion_r1524996902

### CODE REVIEW

File: src/Widgets/ColumnContainerWidget/ColumnsEditorTab.tsx [hunk 1]
```diff
@@ -524,25 +524,23 @@ function adjustNumberOfColumns(
...
-  // merge last columns into one
-  const colsToMerge = takeRight(
-    originalContents,
-    originalContents.length - splitIndexAt,
+  columns.forEach((column, index) =>
+    column.update({
+      content:
+        index !== columns.length - 1
+          ? originalContents[index]
+          : originalContents.slice(index).flat(),
```
Comment (apepper) [inline src/Widgets/ColumnContainerWidget/ColumnsEditorTab.tsx:536]: I suggest to change the condition to `===`:

```suggestion
        index === columns.length - 1
          ? originalContents.slice(index).flat()
          : originalContents[index],
```

File: src/Widgets/DataFormContainerWidget/DataFormContainerWidgetComponent.tsx [hunk 2]
```diff
@@ -145,8 +145,8 @@ function valueFromElement(
   if (element.type === 'file') {
     const files = element.files
     if (element.multiple) return files ? [...files] : []
-
-    return files?.length === 1 ? files[0] : null
+    if (files?.length === 1) return files[0] || null
```
Comment (apepper) [inline src/Widgets/DataFormContainerWidget/DataFormContainerWidgetComponent.tsx:148]: Why is `|| null` needed?

How about?

```suggestion
    if (files?.length === 1) return files[0]!
```

**Review by apepper — COMMENTED** [review]
FEEDBACK ([f71b1e0](https://github.com/Scrivito/scrivito-portal-app/pull/337/commits/f71b1e091abb66d0a27360925f66fd54f5a9ff87))

Nice 🚀!

Only two smaller code improvements possible.

File: src/Widgets/ColumnContainerWidget/ColumnsEditorTab.tsx (hunk 1 above)
Comment (dcsaszar) [reply src/Widgets/ColumnContainerWidget/ColumnsEditorTab.tsx:536]: My intention was to match the logical order of operations:
1. Fitting columns are copied 1:1 first (top line)
2. The remaining columns are merged and copied to the last index (bottom line)

Does this make sense?

**Review by apepper — APPROVED** [review]
LGTM ([f84f2b1](https://github.com/Scrivito/scrivito-portal-app/pull/337/commits/f84f2b1a3cbda8cbbcf5bbb3544a079eea173425))

Nice 🚀!

**Pull request merged**